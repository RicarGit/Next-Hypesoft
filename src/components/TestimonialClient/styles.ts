import styled from "styled-components"

export const Client = styled.div`
  width: inherit;
  display: flex;
  padding: 24px;
  border-radius: 4px;
  gap: 16px;
  background-color: ${({ theme }) => theme.colors.white};
  cursor: pointer;

  &.active {
    background: linear-gradient(0deg, rgba(223, 227, 235, 0.3),
      rgba(223, 227, 235, 0.3)),
      #FFFFFF;
  }
`

export const ClientAvatar = styled.div`
  width: 52px;
  height: 52px;
  border-radius: 50%;
  background-color: ${({ theme }) => theme.colors.gray[500]};
`

export const ClientInfoContainer = styled.div`
  width: 196px;
  height: 52px;
  display: flex;
  flex-direction: column;
  gap: 4px;
`

export const ClientName = styled.p`
  font-size: 18px;
  font-weight: 400;
  line-height: 28px;
  color: ${({ theme }) => theme.colors.primary};
`

export const ClientOccupation = styled.p`
  font-size: 14px;
  font-weight: 400;
  line-height: 20px;
  color: ${({ theme }) => theme.colors.gray[500]};

  span {
    color: ${({ theme }) => theme.colors.secondary};
  }
`