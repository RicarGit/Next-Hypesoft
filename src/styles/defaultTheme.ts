import { DefaultTheme } from "styled-components"

const defaultTheme: DefaultTheme = {
  colors: {
    primary: '#242424',
    secondary: '#ED1CA6',
    sec: '#7D00FF',
    white: '#FFFFFF',
    gray: {
      100: '#DFE3EB',
      500: '#A5A5A5',
    },
  },
  font: {
    family: 'Inter',
    weigths: {
      regular: 400,
      medium: 500,
      semiBold: 600,
      bold: 700
    },
  },
}

export default defaultTheme