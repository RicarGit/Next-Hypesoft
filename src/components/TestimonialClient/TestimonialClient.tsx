import * as S from './TestimonialClient.styles'

type Client = {
  avatar?: string
  name: string
  occupation: string
  company: string
  className?: 'active'
}

export const TestimonialClient = ({ avatar, name, occupation, company, className }: Client) => {
  return (
    <S.Client className={className}>
      {avatar ? 'e.g. set the avatar image or the image will be empty' : <S.ClientAvatar />}
      <S.ClientInfoContainer>

        <S.ClientName>
          {name}
        </S.ClientName>

        <S.ClientOccupation>
          {occupation} <span>{company}</span>
        </S.ClientOccupation>

      </S.ClientInfoContainer>
    </S.Client>
  )
}