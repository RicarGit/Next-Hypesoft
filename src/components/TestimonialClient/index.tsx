import * as S from './styles'

type Client = {
  avatar?: string
  name: string
  occupation: string
  company: string
  className?: 'active'
}

const TestimonialClient = ({ avatar, name, occupation, company, className }: Client) => {
  return (
    <S.Client className={className}>
      {avatar ? 'e.g. set the avatar image or the image will be empty' : <S.ClientAvatar />}
      <S.ClientInfoContainer>

        <S.ClientName>
          {name}
        </S.ClientName>

        <S.ClientOccupation>
          {occupation} <span>{company}</span>
        </S.ClientOccupation>

      </S.ClientInfoContainer>
    </S.Client>
  )
}

export default TestimonialClient