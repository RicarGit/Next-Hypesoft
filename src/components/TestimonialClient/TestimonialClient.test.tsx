import { render } from '@testing-library/react'
import { ThemeProvider } from 'styled-components'
import defaultTheme from 'styles/defaultTheme'
import TestimonialClient from '.'

describe('TestimonialClient', () => {
  it('should match snapshot', () => {
    const { container } = render(
      <ThemeProvider theme={defaultTheme}>
        <TestimonialClient name='' company='' occupation='' />
      </ThemeProvider>
    )

    expect(container.firstChild).toMatchSnapshot()
  })
})