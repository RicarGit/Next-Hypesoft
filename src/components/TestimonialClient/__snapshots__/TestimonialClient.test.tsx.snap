// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TestimonialClient should match snapshot 1`] = `
<div
  class="sc-bczRLJ fpEDZi"
>
  <div
    class="sc-gsnTZi fAxjVk"
  />
  <div
    class="sc-dkzDqf ekTRlg"
  >
    <p
      class="sc-hKMtZM dlYHTv"
    />
    <p
      class="sc-eCYdqJ bXhjJy"
    >
       
      <span />
    </p>
  </div>
</div>
`;
